test/martha/martha_v3.test.js

```diff
@@ -141,155 +159,121 @@ test.serial('martha_v3 resolves a valid DOS-style url', async (t) => {
-    getJsonFromApiStub.onCall(0).resolves(googleSAKeyObject);
+    const bond = bondUrls('dcf-fence');
+
+    getJsonFromApiStub.withArgs(bond.serviceAccountKeyUrl, terraAuth).resolves(googleSAKeyObject);
+
     const response = mockResponse();
-    await marthaV3(mockRequest({
-        body: {
-            url: 'dos://abc/123',
-            fields: ['googleServiceAccount'],
-        },
-    }), response);
+    await marthaV3(
+        mockRequest({ body: { url: 'dos://abc/123', fields: ['googleServiceAccount'] } }),
+        response
+    );
+
     t.is(response.statusCode, 200);
     const result = response.send.lastCall.args[0];
+    t.deepEqual(result, mask(sampleDosMarthaResult(googleSAKeyObject), 'googleServiceAccount'));
+
     sinon.assert.callCount(getJsonFromApiStub, 1); // DRS was not called
-    t.deepEqual(
-        { ...result },
-        mask(sampleDosMarthaResult(googleSAKeyObject), 'googleServiceAccount'),
-    );
-    const requestedBondSAKeyUrl = getJsonFromApiStub.getCall(0).args[0];
-    const saKeyUrlMatches = requestedBondSAKeyUrl.match(bondSAKeyUrlRegEx);
-    t.truthy(saKeyUrlMatches, 'Bond SA key URL called does not match Bond SA key URL regular expression');
-    const expectedSAKeyProvider = 'dcf-fence';
-    const actualSAKeyProvider = saKeyUrlMatches[2];
-    t.is(actualSAKeyProvider, expectedSAKeyProvider);
 });
 
-// BT-236 temporarily cut access token and access endpoint out of the flow
 test.serial('martha_v3 calls the correct endpoints when only the accessUrl is requested', async (t) => {
-    getJsonFromApiStub.onCall(0).resolves(bdcDrsResponse);
+    const {
+        id: objectId, self_uri: drsUri,
+        access_methods: { 0: { access_id: accessId, access_url: { url: s3Url } } }
+    } = kidsFirstDrsResponse;
+    const bond = bondUrls('kids-first');
+    const drs = drsUrls(config.HOST_KIDS_FIRST_PROD, objectId, accessId);
+    const drsAccessUrlResponse = mockS3AccessUrl(s3Url);
+
+    getJsonFromApiStub.withArgs(drs.objectsUrl, null).resolves(kidsFirstDrsResponse);
+    getJsonFromApiStub.withArgs(bond.accessTokenUrl, terraAuth).resolves(bondAccessTokenResponse);
+    getJsonFromApiStub.withArgs(drs.accessUrl, `Bearer ${bondAccessTokenResponse.token}`)
+        .resolves(drsAccessUrlResponse);
+
     const response = mockResponse();
-    await marthaV3(
-        mockRequest(
-            {
-                body: {
-                    url: 'drs://dg.712C/fa640b0e-9779-452f-99a6-16d833d15bd0',
-                    fields: ['accessUrl'],
-                }
-            },
-        ),
-        response,
-    );
+    await marthaV3(mockRequest({ body: { url: drsUri, fields: ['accessUrl'] } }), response);
+
     t.is(response.statusCode, 200);
     const result = response.send.lastCall.args[0];
-    sinon.assert.callCount(getJsonFromApiStub, 1);
-    t.deepEqual(
-        { ...result },
-        { accessUrl: null });
+    t.deepEqual(result, { accessUrl: drsAccessUrlResponse });
```

Comment:
This is a change in the expectation. This was previously using a BDC CIB DRS URI, but I switched it to KidsFirst since BDC is not yet turned on.